### DISCUSSION TITLE
Support passing headers to RequestExecutor via graphQLClient.executeQuery()

### DISCUSSION THREAD

**Discussion by rachanamamillapalli** [Netflix/dgs-framework#661] [Q&A] [answered]
Currently the graphQLClient.executeQuery method only accepts passing query, variables, operation name, request Executor.  
Is there a way we could pass custom headers down to request executor?

**Comment by rmishra3261** [comment]
Yes you can pass custom headers to request executor. You can use below code where in you can pass requestHeaders as method parameters:

 ```
   private RequestExecutor createRestTemplateRequestExecutor(HttpHeaders requestHeaders){
        return (url, headers, body) -> {
            requestHeaders.putAll(headers);
            ResponseEntity<String> exchange = restTemplate.exchange(url, HttpMethod.POST, new HttpEntity<>(body, requestHeaders), String.class);
            return new HttpResponse(exchange.getStatusCodeValue(), exchange.getBody());
        };
```

**Comment by paulbakker** [comment, marked as answer]
The answer above is correct for using the `RequestExecutor`.

We now also have a `WebClientGraphQLClient`. When using that you can provide headers as follows:

```java
WebClient webClient = WebClient.create("http://localhost:8080/graphql");
WebClientGraphQLClient client = MonoGraphQLClient.createWithWebClient(webClient, headers -> headers.add("myheader", "test"));

client.reactiveExecuteQuery("...");
```

**Reply by jay3047** [reply]
Hi @paulbakker , 

Wanted to know  are we suppose to create WebClientGraphQLClient for each request. As i have a use-case where header is being generated from a variable which changes for every request. 

We wants to keep WebClientGraphQLClient singleton  but challenge is with passing dynamic variable at runtime while executing query

**Comment by rachanamamillapalli** [comment]
@paulbakker @berngp  @srinivasankavitha  Any timeline on when the release candidate would be available for consumption?

**Reply by berngp** [reply]
There is already a https://github.com/Netflix/dgs-framework/releases/tag/v4.9.0-rc.1

**Reply by rachanamamillapalli** [reply]
@berngp Its tagged pre-release. I don't think it's available yet?

**Comment by rachanamamillapalli** [comment]
@paulbakker @berngp Sorry for the multiple threads, but would it be possible to release the new version today or provide an ETA? 😅
I want to avoid the re-work of using deprecated features if possible.

**Reply by paulbakker** [reply]
We just released [4.9.0](https://github.com/Netflix/dgs-framework/releases/tag/v4.9.0). It might take a couple of hours before Maven central is synced.

**Reply by rachanamamillapalli** [reply]
Thanks a bunch. Its in sync now
